### PR TITLE
Addition Info NFTS Pallet

### PR DESCRIPTION
**Pull request by filippoweb3**
More information for devs willing to use the pallet to build dApps (uploading images, metadata and attributes)

### CODE REVIEW

**Comment by filippoweb3** [comment]
@jsidorenko  @SaltyCucumber
I added more info about what devs can do to upload images and metadata. Let me know if it makes sense. One question: where can I add CID when minting an item on the chain using the extrinsic tab?

**Comment by VadimSaveljev** [comment]
> @jsidorenko @SaltyCucumber I added more info about what devs can do to upload images and metadata. Let me know if it makes sense. One question: where can I add CID when minting an item on the chain using the extrinsic tab?

when updating collection: setCollectionMetadata
put CID in data field

when updating NFT: setMetadata
put CID in data field

File: docs/learn/learn-nft-pallets.md [hunk 1]
```diff
@@ -117,6 +124,38 @@ You can mint an NFT using the `nfts.mint` extrinsic. You must then specify the f
...
+### Uploading Files and Metadata
+
+When you have a collection ID and an item ID you need to upload
+
+- Open an account on [Pinata](https://www.pinata.cloud/).
+- Follow
+  [these steps](https://docs.pinata.cloud/what-can-i-learn-here/pinning-your-first-file-with-pinata)
+  to upload the file you want to mint.
+- After uploading your file, get the Content Identifier (CID). This is a unique string of letters
+  and numbers proving that the NFT is yours.
```
Comment (VadimSaveljev) [inline docs/learn/learn-nft-pallets.md]: CID doesn't prove anything, it's just a marker for IPFS

File: docs/learn/learn-nft-pallets.md [hunk 2]
```diff
@@ -117,6 +124,38 @@ You can mint an NFT using the `nfts.mint` extrinsic. You must then specify the f
...
+```
+{
+    "name":"Your Collection Name",
+    "description":"Collection's Description",
+    "image":"Your Collection CID"
+}
+```
+
+- Upload the metadata file to Pinata and get the updated CID.
+- You can add the CID when minting your NFT on the Polkadot-JS UI. Go to Developer > Extrinsics and
```
Comment (VadimSaveljev) [inline docs/learn/learn-nft-pallets.md]: `when updating metadata of your NFT`
we already minted it, you can't change metadata if you haven't minted the NFT first

File: docs/learn/learn-nft-pallets.md [hunk 3]
```diff
@@ -150,9 +189,18 @@ Creating an item usually involves setting some attributes specific to that item.
 - Take ownership of an NFT from the [pallet-nfts](#nfts-pallet)
 - Create a new asset in [pallet-assets](https://polkadot.js.org/docs/substrate/extrinsics#assets)
 - Mint the input amount to the previous owner of the NFT as the beneficiary
+- Mass minting: Minting multiple items in one single transaction. Ths will require the user to
```
Comment (VadimSaveljev) [inline docs/learn/learn-nft-pallets.md]: This

File: docs/learn/learn-nft-pallets.md [hunk 4]
```diff
@@ -150,9 +189,18 @@ Creating an item usually involves setting some attributes specific to that item.
...
+  provide a .csv file with two columns: NFT ID and CID if metadata.
```
Comment (VadimSaveljev) [inline docs/learn/learn-nft-pallets.md]: of

File: docs/learn/learn-nft-pallets.md [hunk 5]
```diff
@@ -150,9 +189,18 @@ Creating an item usually involves setting some attributes specific to that item.
...
 
 ## Uniques Pallet
 
+:::info
+
+The Uniques Pallet is being decommissioned. Everything related to NFTs will be covered by the the
```
Comment (VadimSaveljev) [inline docs/learn/learn-nft-pallets.md]: deprecated
decommissioned is a strong word I think in this case, we are not removing it (at least not yet)

File: docs/learn/learn-nft-pallets.md [hunk 6]
```diff
@@ -117,6 +124,45 @@ You can mint an NFT using the `nfts.mint` extrinsic. You must then specify the f
 
 Creating an item usually involves setting some attributes specific to that item.
 
+### Uploading Files and Metadata
+
+When you have a collection ID and an item ID you need to upload
```
Comment (profradha) [inline docs/learn/learn-nft-pallets.md]:
```suggestion
When you have a collection ID and an item ID you need to 
```

File: docs/learn/learn-nft-pallets.md [hunk 7]
```diff
@@ -117,6 +124,45 @@ You can mint an NFT using the `nfts.mint` extrinsic. You must then specify the f
...
+- Upload the metadata file to Pinata and get the updated CID.
+- You can add the CID after minting your NFT on the Polkadot-JS UI. Go to Developer > Extrinsics and
+  select the `nfts.setCollectionMetadata` (for collections) or `nfts.setMetadata` (for single NFTs)
+  extrinsic. Under the `data: Bytes` field you can enter the CID or upload the metadata file.
+
+The collection can be created and its item minted before uploading the NFT file and related
+metadata. The minting process on chain will assign a collection and item ID to your account. Those
+IDs will later be populated with NFT files, metadata, and attributes. Once you uploaded the NFT
+files and related data, the above-mentioned extrinsics can be used to update a collection or item.
+
```
Comment (profradha) [inline docs/learn/learn-nft-pallets.md]:
```suggestion
The collection can be created and its item minted before uploading the NFT file and related
metadata. The minting process on-chain will assign a collection and item ID to your account. Those
IDs will later be populated with NFT files, metadata, and attributes. Once you upload the NFT
files and related data, the extrinsics mentioned above can be used to update a collection or item.

```

File: docs/learn/learn-nft-pallets.md [hunk 8]
```diff
@@ -117,6 +124,45 @@ You can mint an NFT using the `nfts.mint` extrinsic. You must then specify the f
...
+:::info NFT/DEX/Asset Portal
+
+With the new NFT/DEX/Asset portal, all the above steps will be executed "under-the-hood" and the
+user will not have to worry about all technicalities.
+
```
Comment (profradha) [inline docs/learn/learn-nft-pallets.md]:
```suggestion
With the new NFT/DEX/Asset portal, all the above steps will be executed "under the hood" and the
user will not have to worry about all technicalities.

```

**Review by DrW3RK — APPROVED** [review]

**Comment by jsidorenko** [comment]
"and there will be the option to attach https://github.com/paritytech/substrate/pull/12437."
Can we remove this, please? That was just an idea which didn't get into the roadmap

**Comment by filippoweb3** [comment]
> "and there will be the option to attach [paritytech/substrate#12437](https://github.com/paritytech/substrate/pull/12437)." Can we remove this, please? That was just an idea which didn't get into the roadmap

see this [new PR](https://github.com/w3f/polkadot-wiki/pull/4486)

**Pull request merged**